pkg/build/apiserver/admission/secretinjector/admission.go

```diff
@@ -93,7 +104,13 @@ func (si *secretInjector) Admit(attr admission.Attributes) (err error) {
 	if match := urlpattern.Match(patterns, url); match != nil {
 		secretName := match.Cookie.(string)
 		glog.V(4).Infof(`secretinjector: matched secret "%s/%s" to buildconfig "%s"`, namespace, secretName, bc.GetName())
-		bc.Spec.Source.SourceSecret = &api.LocalObjectReference{Name: secretName}
+		if bc.Spec.Source.SourceSecret == nil || bc.Spec.Source.SourceSecret.Name != secretName {
+			if mutationAllowed {
+				bc.Spec.Source.SourceSecret = &api.LocalObjectReference{Name: secretName}
+			} else {
+				return admission.NewForbidden(attr, fmt.Errorf("mutated spec.source.sourceSecret, expected: %v, got %v", api.LocalObjectReference{Name: secretName}, bc.Spec.Source.SourceSecret))
+			}
+		}
```

Comment:
Isn't this changing behavior by suddenly forbidding stuff during Validation that was previously admitted ?

Reply:
It shouldn't be as long as nobody did dirty post-secretInjector mutation in admission webhooks, which we have disabled by default.